src/playdate/system.nim

```diff
@@ -198,6 +198,14 @@ proc getReduceFlashing* (this: ptr PlaydateSys): bool =
     privateAccess(PlaydateSys)
     return this.getReduceFlashing() == 1
 
+var serialMsgCallback: proc(msg: string)
+
+proc privateSerialMessageCallback(msg: ConstChar) {.cdecl, raises: [].} = serialMsgCallback($msg)
+
+proc setSerialMessageCallback*(this: ptr PlaydateSys, callback: proc(msg: string)) =
```

Comment:
I'm not sure this supports un-setting a callback, but perhaps that's not needed or even supported by the C api. I suppose providing a proc that does nothing suffices